src/Core/Components/Dialog/Parameters/DialogParameters.cs

```diff
@@ -167,6 +167,14 @@ public virtual HorizontalAlignment Alignment
     /// This method is only called when using the <see cref="IDialogService"/>.
     /// </remarks>
     public EventCallback<DialogInstance> OnDialogOpened { get; set; } = default!;
+
+    /// <summary>
+    /// Function that is called and awaited before the dialog is closed.
+    /// </summary>
+    /// <remarks>
+    /// This is a suitable callback to use when you need to validate the data in the dialog <em>before</em> it closes.
+    /// </remarks>
+    public Func<Task<bool>> OnDialogValidation { get; set; } = default!;
```

Comment:
Please change the name to `ValidateDialogAsync`. As discussed it is not an `EventCallback` and Async needs to be added because it returns a Task<bool>, indicating that it is an asynchronous method.